neptune-tinkerpop/3.4.13/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/step/util/HasContainer.java

```diff
@@ -0,0 +1,227 @@
+        if (this.key != null) {
+            if (this.key.equals(T.id.getAccessor()))
+                return testingIdString ? testIdAsString(element) : testId(element);
+            if (this.key.equals(T.label.getAccessor()))
+                return testLabel(element);
+        }
+
+        final Iterator<? extends Property> itty = element.properties(this.key);
+        try {
+            while (itty.hasNext()) {
+                if (testValue(itty.next()))
+                    return true;
+            }
+        } finally {
+            CloseableIterator.closeIterator(itty);
+        }
+        return false;
+    }
+
+    public final boolean test(final Property property) {
+        if (this.key != null) {
+            if (this.key.equals(T.value.getAccessor()))
+                return testValue(property);
+            if (this.key.equals(T.key.getAccessor()))
+                return testKey(property);
+        }
+        if (property instanceof Element)
+            return test((Element) property);
+        return false;
+    }
+
+    protected boolean testId(final Element element) {
+        return this.predicate.test(element.id());
+    }
+
+    protected boolean testIdAsString(final Element element) {
+        return this.predicate.test(element.id().toString());
+    }
+
+    protected boolean testLabel(final Element element) {
+
+        /* start patch for localstack */
+
+        // add comparison predicates to support the multi-label syntax `label1::label2::label3`
+
+        class LabelEquals implements BiPredicate<String, String> {
+            public boolean test(String label, String otherLabel) {
+                if (label == null) return false;
+                for (String partialLabel: label.split(LABEL_DELIMITER)) {
+                    if (partialLabel.equals(otherLabel)) {
+                        return true;
+                    }
+                }
+                return false;
+            }
+        }
+
+        class LabelWithin implements BiPredicate<String, String[]> {
+            public boolean test(String label, String[] labels) {
+                if (label == null) return false;
+                final List<String> labelsList = Arrays.asList(labels);
+                for (String partialLabel: label.split(LABEL_DELIMITER)) {
+                    if (labelsList.contains(partialLabel)) {
+                        return true;
+                    }
+                }
+                return false;
+            }
+        }
+
+        if (this.predicate.getBiPredicate() == Compare.eq && this.predicate.getValue() instanceof String) {
+            this.predicate = new P(new LabelEquals(), this.predicate.getValue());
+        } else if (this.predicate.getBiPredicate() == Contains.within && this.predicate.getValue() instanceof String[]) {
+            this.predicate = new P(new LabelWithin(), this.predicate.getValue());
+        }
+
+        /* end patch for localstack */
```

Comment:
This part here is the patched code - rest of this class is copied from original Tinkerpop source code..